package/MDAnalysis/coordinates/dcdtimeseries.pyx

```diff
@@ -109,18 +111,25 @@ def __read_timecorrel(object self, object atoms, object atomcounts, object forma
     cdef int index, numskip
     cdef int i, j
     cdef float unitcell[6]
+    cdef int remaining_frames = stop-start
+    numskip = 0
+    print n_frames
     for i from 0 <= i < n_frames:
-        if (skip > 1):
+        if (step > 1 and i > 0):
             # Check if we have fixed atoms
             # XXX not done
-            numskip = skip - (dcd.setsread % skip) - 1
+            numskip = step - 1
+            if(remaining_frames < numskip):
+               numskip = 1
+
             rc = skip_dcdstep(dcd.fd, dcd.natoms, dcd.nfixed, dcd.charmm, numskip)
             if (rc < 0):
                 raise IOError("Error skipping frame from DCD file")
-            dcd.setsread = dcd.setsread + numskip
+        dcd.setsread = dcd.setsread + numskip
```

Comment:
Are you sure that this should now be one level out, i.e. not *inside* the `if` statement any more?